src/features/user/DonationReceipt/DonorContactManagement.tsx

```diff
@@ -0,0 +1,183 @@
+import { ErrorHandlingContext } from '../../common/Layout/ErrorHandlingContext';
+
+type StoredReceiptData = {
+  dtn: string;
+  year: string;
+  challenge: string;
+};
+
+const DonorContactManagement = () => {
+  const t = useTranslations('DonationReceipt');
+  const router = useRouter();
+  const {
+    donationReceiptData,
+    setDonationReceiptData,
+    updateDonationReceiptData,
+  } = useDonationReceipt();
+  const { user, token, contextLoaded, logoutUser } = useUserProps();
+  const { tenantConfig } = useTenant();
+  const { setErrors } = useContext(ErrorHandlingContext);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [addressAction, setAddressAction] = useState<AddressAction | null>(
+    null
+  );
+  const [selectedAddressForAction, setSelectedAddressForAction] =
+    useState<Address | null>(null);
+  const [donorAddresses, setDonorAddresses] = useState<Address[]>(
+    user?.addresses ?? []
+  );
+  const [isLoading, setIsLoading] = useState(false);
+
+  const navigateToVerificationPage = useCallback(() => {
+    if (donationReceiptData) {
+      const { dtn, challenge, year } = donationReceiptData;
+      router
+        .push(
+          `/verify-receipt-data?dtn=${dtn}&challenge=${challenge}&year=${year}`
+        )
+        .then(() => setIsLoading(false));
+    }
+  }, [donationReceiptData, router]);
+
+  useEffect(() => {
+    if (donationReceiptData) return;
+    const receiptDataString = sessionStorage.getItem('receiptData');
+    const parsedData: StoredReceiptData = receiptDataString
+      ? JSON.parse(receiptDataString)
+      : null;
+    if (!parsedData) {
+      router.push('/');
+      return;
+    }
+    const { dtn, year, challenge } = parsedData;
+    const fetchReceiptData = async () => {
+      try {
+        const data = await getRequest<ReceiptDataAPI>({
+          tenant: tenantConfig.id,
+          url: '/app/donationReceipt',
+          queryParams: {
+            dtn,
+            year,
+            challenge,
+          },
+        });
+        if (data) {
+          updateDonationReceiptData(data);
+          sessionStorage.removeItem('receiptData');
+        }
+      } catch (err) {
+        setErrors(handleError(err as APIError));
+        router.push('/');
+      }
+    };
+
+    fetchReceiptData();
+  }, [
+    tenantConfig.id,
+    updateDonationReceiptData,
+    handleError,
+    donationReceiptData,
+    router,
+  ]);
+
+  const updateDonorAddresses = useCallback(async () => {
+    if (!user || !token || !contextLoaded) return;
+    try {
+      const res = await getAuthenticatedRequest<Address[]>({
+        tenant: tenantConfig.id,
+        url: '/app/addresses',
+        token,
+        logoutUser,
+      });
+      if (res) setDonorAddresses(res);
+    } catch (error) {
+      setErrors(handleError(error as APIError));
+    }
+  }, [user, token, contextLoaded, tenantConfig.id, logoutUser, setErrors]);
+
+  useEffect(() => {
+    updateDonorAddresses();
```

Comment:
You probably don't need this initially, as the profile has the `addresses` property.

Reply:
We need this logic "`For example`, if a user visits the donor contact management page and adds a new address, then returns to the verification page, and subsequently goes back to the donor contact management page, the `donorAddresses `state gets set to the address from the profile initially. As a result, we won't see the updated address."